### DISCUSSION TITLE
revoke() logic

### DISCUSSION THREAD

**Discussion by owlen** [abdelhamidbakhta/token-vesting-contracts#79] [General]
Hi All!
Can anyone (@abdelhamidbakhta ?) confirm the thinking behind the revoke() function?

I assumed it is used to block the ability of the beneficiary to release tokens, but it release()s the vested tokens. Is the idea is to give the beneficiary whatever vested until now, and block his ability to vest future tokens?

**Comment by AbdelStark** [comment]
> Hi All! Can anyone (@abdelhamidbakhta ?) confirm the thinking behind the revoke() function?
> 
> I assumed it is used to block the ability of the beneficiary to release tokens, but it release()s the vested tokens. Is the idea is to give the beneficiary whatever vested until now, and block his ability to vest future tokens?

Yep you perfectly understood.
